src/domain/wallets/tx-history.ts

```diff
@@ -0,0 +1,189 @@
+import { toSats } from "@domain/bitcoin"
+import { LedgerTransactionType } from "@domain/ledger"
+import { MEMO_SHARING_SATS_THRESHOLD } from "@config/app"
+import { SettlementMethod } from "./settlement-method"
+
+const addPendingIncoming = (
+  confirmedTransactions: WalletTransaction[],
+  pendingTransactions: SubmittedTransaction[],
+  addresses: OnChainAddress[],
+  usdPerSat: UsdPerSat,
+): WalletTransactionHistoryWithPending => {
+  const walletTransactions: WalletTransaction[] = []
+  pendingTransactions.forEach(({ id, rawTx, createdAt }) => {
+    rawTx.outs.forEach(({ sats, address }) => {
+      if (addresses.includes(address)) {
+        walletTransactions.push({
+          id,
+          settlementVia: SettlementMethod.OnChain,
+          old: {
+            description: "pending",
+            usd: usdPerSat * sats,
+            feeUsd: 0,
+            type: LedgerTransactionType.OnchainReceipt,
+          },
+          settlementFee: toSats(0),
+          pendingConfirmation: true,
+          createdAt: createdAt,
+          settlementAmount: sats,
+          addresses: [address],
+        })
+      }
+    })
+  })
+  return {
+    transactions: [...walletTransactions, ...confirmedTransactions],
+  }
+}
+
+export const fromLedger = (
+  ledgerTransactions: LedgerTransaction[],
+): ConfirmedTransactionHistory => {
+  const transactions = ledgerTransactions.map(
+    ({
+      id,
+      memoFromPayer,
+      lnMemo,
+      type,
+      credit,
+      debit,
+      fee,
+      usd,
+      feeUsd,
+      paymentHash,
+      username,
+      addresses,
+      pendingConfirmation,
+      timestamp,
+    }) => {
+      const settlementAmount = toSats(credit - debit)
+      const description = translateDescription({
+        type,
+        memoFromPayer,
+        lnMemo,
+        credit,
+        username,
+      })
+      if (type === LedgerTransactionType.OnchainIntraLedger && addresses) {
+        return {
+          id,
+          settlementVia: SettlementMethod.IntraLedger,
+          old: {
+            description,
+            usd,
+            feeUsd,
+            type,
+          },
+          settlementAmount,
+          settlementFee: fee,
+          paymentHash: null,
+          addresses: addresses,
+          recipientId: null,
+          pendingConfirmation,
+          createdAt: timestamp,
+        }
+      }
+      if (type === LedgerTransactionType.IntraLedger) {
+        return {
+          id,
+          settlementVia: SettlementMethod.IntraLedger,
+          old: {
+            description,
+            usd,
+            feeUsd,
+            type,
+          },
+          settlementAmount,
+          settlementFee: fee,
+          paymentHash: (paymentHash as PaymentHash) || null,
+          recipientId: username || null,
+          addresses: null,
+          pendingConfirmation,
+          createdAt: timestamp,
+        }
+      } else if (addresses && addresses.length > 0) {
+        return {
```

Comment:
Probably doesn't need an `else` here because of early return in previous block and consistency with Ln 86.

```suggestion
      }
      if (addresses && addresses.length > 0) {
        return {
```